src/data/notification.rs

```diff
@@ -11,12 +12,68 @@ pub struct Notification {
     pub body: String,
     pub expire_timeout: Timeout,
     pub urgency: Urgency,
+    pub category: Category,
     pub image_data: Option<ImageData>,
     pub image_path: Option<String>,
     pub is_read: bool,
     pub created_at: u64,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub enum Category {
+    Device(Option<CategoryEvent>),
+    Email(Option<CategoryEvent>),
+    Im(Option<CategoryEvent>),
+    Network(Option<CategoryEvent>),
+    Presence(Option<CategoryEvent>),
+    Transfer(Option<CategoryEvent>),
+    #[default]
+    None,
+}
```

Comment:
I recommend to add `CategoryEvent::Generic` and use it instead of `Option<CategoryEvent>`, because it is easy to understand what means.